src/modm/platform/core/e31/linkerscript.ld.in

```diff
@@ -0,0 +1,248 @@
+	} >FLASH
+
+
+	/* initialized variables */
+	.data : ALIGN(4)
+	{
+		__data_load = LOADADDR(.data);
+		__data_start = .;
+		*(.data .data.* .gnu.linkonce.d.*)
+		. = ALIGN(4);
+		__data_end = .;
+	} >RAM AT >FLASH
+
+	.stack (NOLOAD) :
+	{
+		__stack_start = .;
+
+		. += MAIN_STACK_SIZE;
+		. = ALIGN(8);
+		__main_stack_top = .;
+
+		. += PROCESS_STACK_SIZE;
+		. = ALIGN(8);
+		__process_stack_top = .;
+
+		__stack_end = .;
+	} >RAM
+
+	/* uninitialized variables */
+	.bss (NOLOAD) :
+	{
+		__bss_start = . ;
+		*(.bss .bss.* .gnu.linkonce.b.*)
+		. = ALIGN(4);
+		__bss_end = .;
+	} >RAM
+
+	/* Global data not cleared after reset.  */
+	.noinit (NOLOAD) :
+	{
+		__noinit_start = .;
+		*(.noinit*)
+		. = ALIGN(4);
+		__noinit_end = .;
+	} >RAM
+
+
+	.fastdata : ALIGN(4)
+	{
+		__fastdata_load = LOADADDR(.fastdata);
+		__fastdata_start = .;
+		*(.fastdata)
+		. = ALIGN(4);
+		__fastdata_end = .;
+	} >RAM AT >FLASH
+
+
+	.heap1 (NOLOAD) :
+	{
+		__heap1_start = .;
+		. = ORIGIN(RAM) + LENGTH(RAM);
+		__heap1_end = .;
+	} >RAM
+
+
+
+
+	/* TABLES! TABLES! ALL THE TABLES YOU COULD EVER WANT! TABLES! */
+	.table.zero.intern : ALIGN(4)
+	{
+		__table_zero_intern_start = .;
+		LONG(__bss_start)
+		LONG(__bss_end)
+		__table_zero_intern_end = .;
+	} >FLASH
+
+
+	.table.copy.intern : ALIGN(4)
+	{
+		__table_copy_intern_start = .;
+		LONG(__data_load)
+		LONG(__data_start)
+		LONG(__data_end)
+		LONG(__fastdata_load)
+		LONG(__fastdata_start)
+		LONG(__fastdata_end)
+		__table_copy_intern_end = .;
+	} >FLASH
+
+
+	.table.zero.extern : ALIGN(4)
+	{
+		__table_zero_extern_start = .;
+
+		__table_zero_extern_end = .;
+	} >FLASH
+
+	.table.copy.extern : ALIGN(4)
+	{
+		__table_copy_extern_start = .;
+
+		__table_copy_extern_end = .;
+	} >FLASH
+
+
+	/* SRAM properties bit mask (16-bit):
+	 *
+	 * - 0x0001: accessible via S-Bus
+	 * - 0x0002: accessible via D-Bus
+	 * - 0x0004: accessible via I-Bus
+	 * - 0x0008: accessible via DMA
+	 * - 0x0010: accessible via DMA2D
+	 *
+	 * - 0x1FE0: reserved
+	 *
+	 * - 0x2000: Fast memory (Core- or Tightly-Coupled)
+	 * - 0x4000: non-volatile (or battery backed) memory
+	 * - 0x8000: external memory
+	 */
+	.table.heap : ALIGN(4)
+	{
+		__table_heap_start = .;
+		LONG(0x000f)
+		LONG(__heap1_start)
+		LONG(__heap1_end)
+
+		__table_heap_end = .;
+	} >FLASH
+
+
+	. = ALIGN(4);
+	__rom_end = .;
+
+
+	/* DWARF debug sections */
+	.debug_abbrev   0 : { *(.debug_abbrev) }
+	.debug_aranges  0 : { *(.debug_aranges) }
+	.debug_frame    0 : { *(.debug_frame) }
+	.debug_info     0 : { *(.debug_info) }
+	.debug_line     0 : { *(.debug_line) }
+	.debug_loc      0 : { *(.debug_loc) }
+	.debug_macinfo  0 : { *(.debug_macinfo) }
+	.debug_pubnames 0 : { *(.debug_pubnames) }
+	.debug_pubtypes 0 : { *(.debug_pubtypes) }
+	.debug_ranges   0 : { *(.debug_ranges) }
+	.debug_str      0 : { *(.debug_str) }
+
+	.comment 0 : { *(.comment) }
+	.ARM.attributes 0 : { KEEP(*(.ARM.attributes)) }
+	/DISCARD/ : { *(.note.GNU-stack)  }
+
+}
```

Comment:
Please use the Jinja macros to generate the sections, this allows me to refactor more easily.
Something like this:
https://github.com/modm-io/modm/blob/develop/src/modm/platform/core/stm32/linkerscript/stm32_ram.ld.in

Reply:
Please note that the stack alignment is due to interrupt table alignment requirements on Cortex-M and *only* for remapping the table into ROM into RAM. This is probably(?) not required for the E31.